oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticDocumentMaker.java

```diff
@@ -138,22 +142,33 @@ protected boolean isFulltextValuePersistedAtNode(PropertyDefinition pd) {
 
     @Override
     protected void indexTypedProperty(ElasticDocument doc, PropertyState property, String pname, PropertyDefinition pd, int i) {
-        int tag = property.getType().tag();
-
+        // Get the Type tag from the defined index definition here - and not from the actual persisted property state - this way in case
+        // If the actual property value is different from the property type defined in the index definition/mapping - this will try to convert the property if possible,
+        // other wise will log a warning and not try and add the property to index. If we try and index incompatible data types (like String to Date),
```

Comment:
typo
```suggestion
        // otherwise will log a warning and not try and add the property to index. If we try and index incompatible data types (like String to Date),
```